### PR TITLE
Provide ability to reset password if forgotten

### PR DESCRIPTION
**Pull request by jhsi517**
## Pull Request Information
**Relates to Github Issue:** #59 

**What is the PR Type:** feature

**Describe the added/fixed behaviour:**
- Have set up a SMTP server that will send E-mail to the user that has submitted a valid request for a password reset.
- Ability to reset password once user access a unique link via their E-mail inbox.
    - Set up the following 3 APIs to achieve the feature above.
        - POST user/resetTokenCheck
        - POST user/resetPassword
        - POST user/forgotPassword
- Added 2 more columns in user database/model, "ResetToken (String)" and "CanReset (Bool)"
    - ResetToken is a base 64 string time based token that gets generated when user has forgotten their password. It gets sets to empty once it's used or it will expire after 1 hour (from when the E-mail is sent)
    - CanReset is to check if the user is in a state that can change password. Is implemented to prevent people changing query string to reset other accounts' password.

## Checklist

- [x] Pull request has suitable title, adequately described and associated with a github issue
- [x] Branch name follows correct convention (e.g. feature/[id]-some-feature)
- [x] Branch builds and executes with no errors
- [ ] Test cases has been generated for the feature and test cases pass
- [x] Updated relevant wiki/documentation

## Additional Information
Please either sign up with a legit email address or change one of the test users' email in DevelopmentDatabaseSetup to test the full functionality of this. Because the reset password page does require a unique URL.

This PR mainly provides the backend of the feature. Frontend was developed by @gargi-bansal and have been merged into master in PR #122 and #149 

## Final Reminders
Please remember to squash the commits

### CODE REVIEW

File: WebApiBackend/WebApiBackend/Migrations/FlatManagementContextModelSnapshot.cs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                    b.Property<string>("Name")
+                        .HasColumnType("TEXT");
+
+                    b.HasKey("Id");
+
+                    b.ToTable("TestItems");
+                });
+
+            modelBuilder.Entity("WebApiBackend.Model.User", b =>
+                {
```
Comment (JedLJRobertson) [inline WebApiBackend/WebApiBackend/Migrations/FlatManagementContextModelSnapshot.cs]: Hey Lucas, we're not using migrations at all. Could you clean these out?

The database should autopopulate on launch, and is reset everytime (in future I guess the application would need something for prod, but this is suitable for the dev phase).

Comment (JedLJRobertson) [reply WebApiBackend/WebApiBackend/Migrations/FlatManagementContextModelSnapshot.cs]: I'll run this and test it ASAP :).

Comment (jhsi517) [reply WebApiBackend/WebApiBackend/Migrations/FlatManagementContextModelSnapshot.cs]: Have removed the folder. Please confirm :)

**Comment by jhsi517** [comment]
A wiki page on how to use the SMTP server has been created:
https://github.com/SE701ProjectGroup4/Flatmate-Management-System/wiki/SMTP-Server-Account-and-Usage

File: Frontend/frontend/src/App.js [hunk 2]
```diff
@@ -19,6 +19,8 @@ import APIRequest from './Util/APIRequest';
 import NewFlat from './Components/NewFlat';
 import MembersPage from './Components/Members';
 import ForgotPassword from './Views/ForgotPassword';
+import ResetPassword from './Views/ResetPassword';
+import MembersPage from './Components/Members';
```
Comment (JedLJRobertson) [inline Frontend/frontend/src/App.js]: Yo, this doesn't build for me because MembersPage is duplicated.

Comment (jhsi517) [reply Frontend/frontend/src/App.js]: Thank you for pointing this out. Have removed duplicate and should now render correctly :)

**Pull request merged**